plugins/groovy/groovy-psi/src/org/jetbrains/plugins/groovy/codeInspection/changeToOperator/data/MethodCallData.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2000-2016 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jetbrains.plugins.groovy.codeInspection.changeToOperator.data;
+
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.tree.IElementType;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.plugins.groovy.lang.psi.GroovyPsiElementFactory;
+import org.jetbrains.plugins.groovy.lang.psi.api.statements.expressions.*;
+import org.jetbrains.plugins.groovy.lang.psi.impl.utils.ParenthesesUtils;
+
+import static org.jetbrains.plugins.groovy.lang.psi.impl.utils.BoolUtils.isNegation;
+import static org.jetbrains.plugins.groovy.lang.psi.impl.utils.ComparisonUtils.isComparison;
+
+public final class MethodCallData {
+  private final PsiElement backingElement;
+
+  private boolean isNegated;
+  private GrExpression base;
+  private GrExpression[] arguments;
+  private IElementType comparison;
+
+  @Nullable
+  public static MethodCallData create(@NotNull PsiElement backingElement) {
+    return Builder.build(backingElement);
+  }
+
+  private MethodCallData(@NotNull PsiElement backingElement) {
+    this.backingElement = backingElement;
+  }
+
+  @NotNull
+  public PsiElement getBackingElement() {
+    return backingElement;
+  }
+
+  public boolean isNegated() {
+    return isNegated;
+  }
+
+  @NotNull
+  public String getBase() {
+    return base.getText();
+  }
+
+  @NotNull
+  public String getArgument(int i) {
+    return arguments[i].getText();
```

Comment:
I get *ArrayIndexOutOfBoundsException* of errors, when I try to type `a.isCase(<caret>)` in editor